v1/edgegw_alb_virtual_service_types.go

```diff
@@ -0,0 +1,63 @@
+package v1
+
+import (
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+	govcdtypes "github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+type (
+
+	// EdgeGatewayALBVirtualService represents a virtual service on an NSX-T Edge Gateway.
+	// It contains the Edge Gateway Object, the Virtual Service Model and the NSX-T ALB Virtual Service.
+	EdgeGatewayALBVirtualService struct {
+		client         *EdgeClient
+		VirtualService *EdgeGatewayALBVirtualServiceModel
+		nsxtALBVS      *govcd.NsxtAlbVirtualService
+	}
+
+	EdgeGatewayALBVirtualServiceModel struct {
+		ID string `json:"id,omitempty"`
+
+		// Name contains meaningful name
+		Name string `json:"name,omitempty"`
+
+		// Description is optional
+		Description string `json:"description,omitempty"`
+
+		// Enabled defines if the virtual service is enabled to accept traffic
+		Enabled *bool `json:"enabled"`
+
+		// ApplicationProfile sets protocol for load balancing by using NsxtAlbVirtualServiceApplicationProfile
+		ApplicationProfile govcdtypes.NsxtAlbVirtualServiceApplicationProfile `json:"applicationProfile"`
```

Comment:
ApplicationProfile is a list of 4 type `HTTP`, `HTTPS`, `L4`, `L4_TLS`. 
For this create a new golang type with 4 constants and 1 var which contains 4 consts. 
The variable can be used to validate in the terraform provider for example

Ex: 
```go
type EdgeGatewayALBVirtualServiceModelApplicationProfile string

const (
	EdgeGatewayALBVirtualServiceModelApplicationProfileHTTP EdgeGatewayALBVirtualServiceModelApplicationProfile = "HTTP"
	EdgeGatewayALBVirtualServiceModelApplicationProfileHTTPS EdgeGatewayALBVirtualServiceModelApplicationProfile = "HTTPS"
	[...]
)
var EdgeGatewayALBVirtualServiceModelApplicationProfiles = []EdgeGatewayALBVirtualServiceModelApplicationProfile{
	EdgeGatewayALBVirtualServiceModelApplicationProfileHTTP,
	EdgeGatewayALBVirtualServiceModelApplicationProfileHTTPS
}
```